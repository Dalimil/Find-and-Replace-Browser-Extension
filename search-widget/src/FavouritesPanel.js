import React from 'react';

import FontAwesome from 'react-fontawesome';
import Storage from './Storage';

class FavouritesPanel extends React.Component {
  constructor(props) {
    super(props);

    this.handleFavouriteRemoved = this.handleFavouriteRemoved.bind(this);
    this.handleFavouriteSelected = this.handleFavouriteSelected.bind(this);
  }

  handleFavouriteSelected(id) {
    // todo- callback up - fill in state and close
  }

  handleFavouriteRemoved(id) {
    Storage.setInFavourites(this.props.favourites[id], /* delete */ true);
  }

  render() {
    return (
      <div className="favourites-list">
        <div className="panel-title">Favourites</div>
        <div>
          {Object.keys(this.props.favourites).map(id => {
            const { findTextInput, replaceTextInput } = this.props.favourites[id];
            return (
              <div className="favourites-list-item" key={id}
                  onClick={() => this.handleFavouriteSelected(id)}>
                <span>
                  {findTextInput}
                  <FontAwesome name='long-arrow-right' />
                  {replaceTextInput}
                </span>
                <FontAwesome className="favourites-list-item-remove" name='times'
                  onClick={() => this.handleFavouriteRemoved(id)} />
              </div>
            );
          })}
        </div>
      </div>
    );
  }

}

export default FavouritesPanel;
