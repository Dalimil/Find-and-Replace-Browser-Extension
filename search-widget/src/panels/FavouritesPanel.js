import React from 'react';

import FontAwesome from 'react-fontawesome';
import Storage from '../Storage';

class FavouritesPanel extends React.Component {
  constructor(props) {
    super(props);

    this.handleFavouriteRemoved = this.handleFavouriteRemoved.bind(this);
    this.handleFavouriteSelected = this.handleFavouriteSelected.bind(this);
  }

  handleFavouriteSelected(id) {
    this.props.onFavouriteSelected(this.props.favourites[id]);
  }

  handleFavouriteRemoved(id) {
    Storage.setInFavourites(this.props.favourites[id], /* delete */ true);
  }

  render() {
    const noSavedFavouritesMessage = <div style={{ padding: '1em' }}>
        Currently you have no saved items.</div>;

    return (
      <div className="favourites-list">
        <div className="panel-title"><FontAwesome name='star' fixedWidth={true} /> Favourites</div>
        <div>
          {Object.keys(this.props.favourites).length == 0 && noSavedFavouritesMessage}
          {Object.keys(this.props.favourites).sort().map(id => {
            const { findTextInput, replaceTextInput } = this.props.favourites[id];
            return (
              <div className="favourites-list-item" key={id}
                  onClick={() => this.handleFavouriteSelected(id)}>
                <span>
                  <span>{findTextInput}</span> <FontAwesome name='long-arrow-right' /> <span>{replaceTextInput}</span>
                </span>
                <span>
                  <FontAwesome className="favourites-list-item-remove" name='times'
                    onClick={(e) => {
                      e.stopPropagation();
                      this.handleFavouriteRemoved(id);
                    }} />
                </span>
              </div>
            );
          })}
        </div>
      </div>
    );
  }

}

export default FavouritesPanel;
