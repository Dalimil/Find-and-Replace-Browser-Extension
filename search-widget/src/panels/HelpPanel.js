import React from 'react';

import FontAwesome from 'react-fontawesome';
import { Toggle } from '../InputElements';

import Storage from '../Storage';
import Analytics from '../Analytics';

class HelpPanel extends React.Component {
  constructor(props) {
    super(props);

    this.navigate = this.navigate.bind(this);
    this.baseUrl = 'https://find-and-replace-f6588.firebaseapp.com/';
    this.feedbackUrl = this.baseUrl + 'feedback';
    this.contributeUrl = 'https://www.paypal.me/Dalimil/5USD';
    this.helpUrl = '/help.html';
    this.privacyPolicyUrl = 'https://www.iubenda.com/privacy-policy/8243040';

    this.state = {
      analyticsEnabled: true
    };
    this.handleAnalyticsToggled = this.handleAnalyticsToggled.bind(this);
  }

  componentDidMount() {
    Storage.isAnalyticsEnabled().then(enabled => {
      this.setState({ analyticsEnabled: enabled });
    });
  }

  handleAnalyticsToggled(e) {
    const enabled = e.target.checked;
    this.setState({ analyticsEnabled: enabled });
    if (!enabled) {
      Analytics.sendEvent("analytics-switch", "analytics-disabled");
    }
    Storage.setAnalyticsEnabled(enabled);
    if (enabled) {
      Analytics.sendEvent("analytics-switch", "analytics-enabled");
    }
  }

  navigate(e) {
    e.preventDefault();
    const href = e.target.href;
    chrome.tabs.create({ url: href });
  }

  render() {
    return (
      <div className="help-panel">
        <div className="help-panel-title"><FontAwesome name='question-circle' fixedWidth={true} /> Help</div>
        <img className="help-panel-icon" src='/images/icon48.png' />
        <div style={{ fontWeight: 'bold' }} >Find &amp; Replace for Text Editing</div>
        <div className="help-panel-links">
          <a onClick={this.navigate} href={this.feedbackUrl} >Feedback</a>
          <a onClick={this.navigate} href={this.baseUrl} >Website</a>
          <a onClick={this.navigate} href={this.contributeUrl} >Contribute</a>

          <a onClick={this.navigate} href={this.helpUrl} >Help</a>
        </div>
        <div className="help-panel-footnote">
          Made with <FontAwesome name='heart-o' /> by Dalimil. Thank you for your support!
        </div>
        <div className="help-panel-footnote">
          <a onClick={this.navigate} href={this.privacyPolicyUrl}>Privacy Policy</a>
          Usage reporting:
          <Toggle
            style={{ transform: 'scale(0.75)', marginTop: '-4px', marginLeft: '-5px' }}
            checked={this.state.analyticsEnabled}
            onChange={this.handleAnalyticsToggled} />
        </div>
      </div>
    );
  }

}

export default HelpPanel;
