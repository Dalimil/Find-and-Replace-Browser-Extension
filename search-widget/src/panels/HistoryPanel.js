import React from 'react';

import FontAwesome from 'react-fontawesome';

class HistoryPanel extends React.Component {
  constructor(props) {
    super(props);

    this.handleHistorySelected = this.handleHistorySelected.bind(this);
  }

  handleHistorySelected(historyItem) {
    this.props.onHistorySelected(historyItem);
  }

  render() {
    const noSavedHistoryMessage = <div style={{ padding: '1em' }}>
        Currently there is no search history.</div>;

    return (
      <div className="history-list">
        <div className="panel-title"><FontAwesome name='history' fixedWidth={true} /> History</div>
        <div>
          {this.props.history.length == 0 && noSavedHistoryMessage}
          {this.props.history.map((historyItem, index) => {
            const { findTextInput, replaceTextInput } = historyItem;
            return (
              <div className="history-list-item" key={index}
                  onClick={() => this.handleHistorySelected(historyItem)}>
                <span>
                  <span>{findTextInput}</span> <FontAwesome name='long-arrow-right' /> <span>{replaceTextInput}</span>
                </span>
              </div>
            );
          })}
        </div>
      </div>
    );
  }

}

export default HistoryPanel;
