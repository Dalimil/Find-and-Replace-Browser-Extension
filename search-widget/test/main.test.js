import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../src/Main';
import HelpPanel from '../src/panels/HelpPanel';
import TemplatesPanel from '../src/panels/TemplatesPanel';
import HistoryPanel from '../src/panels/HistoryPanel';
import FavouritesPanel from '../src/panels/FavouritesPanel';

test('HelpPanel renders correctly', () => {
  const tree = renderer.create(<HelpPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('TemplatesPanel renders correctly', () => {
  const tree = renderer.create(<TemplatesPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('HistoryPanel renders correctly', () => {
  const history = [
    { findTextInput: 'abc', replaceTextInput: 'cdf' },
    { findTextInput: 'abcd', replaceTextInput: 'efgh' },
    { findTextInput: 'car', replaceTextInput: 'boat' }
  ];
  const component = renderer.create(<HistoryPanel history={history} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.history = [];
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FavouritesPanel renders correctly', () => {
  const favourites = {
    'a': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'b': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'c': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'd': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'e': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'f': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'g': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'h': { findTextInput: 'very long text specified as find input', replaceTextInput: 'very long replace input' },
    'i': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'j': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'k': { findTextInput: 'abc', replaceTextInput: 'cdf' },
    'l': { findTextInput: 'abc', replaceTextInput: 'cdf' }
  };
  const component = renderer.create(<FavouritesPanel favourites={favourites} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.favourites = {};
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Whole widget renders correctly', () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});