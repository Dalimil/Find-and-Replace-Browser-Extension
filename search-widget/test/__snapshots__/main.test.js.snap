// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesPanel renders correctly 1`] = `
<div
  className="favourites-list"
>
  <div
    className="panel-title"
  >
    <span
      aria-hidden={true}
      className="fa fa-star fa-fw"
    />
     Favourites
  </div>
  <div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          very long text specified as find input
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          very long replace input
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`FavouritesPanel renders correctly 2`] = `
<div
  className="favourites-list"
>
  <div
    className="panel-title"
  >
    <span
      aria-hidden={true}
      className="fa fa-star fa-fw"
    />
     Favourites
  </div>
  <div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          very long text specified as find input
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          very long replace input
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="favourites-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-times favourites-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`HelpPanel renders correctly 1`] = `
<div
  className="help-panel"
>
  <div
    className="help-panel-title"
  >
    <span
      aria-hidden={true}
      className="fa fa-question-circle fa-fw"
    />
     Help
  </div>
  <img
    className="help-panel-icon"
    src="/images/icon48.png"
  />
  <div
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Find & Replace for Text Editing
  </div>
  <div
    className="help-panel-links"
  >
    <a
      href="https://find-and-replace-f6588.firebaseapp.com/feedback"
      onClick={[Function]}
    >
      Feedback
    </a>
    <a
      href="https://find-and-replace-f6588.firebaseapp.com/"
      onClick={[Function]}
    >
      Website
    </a>
    <a
      href="https://www.paypal.me/Dalimil/5USD"
      onClick={[Function]}
    >
      Contribute
    </a>
    <a
      href="/help.html"
      onClick={[Function]}
    >
      Help
    </a>
  </div>
  <div
    className="help-panel-footnote"
  >
    Made with 
    <span
      aria-hidden={true}
      className="fa fa-heart-o"
    />
     by Dalimil. Thank you for your support!
  </div>
  <div
    className="help-panel-footnote"
  >
    <a
      href="https://www.iubenda.com/privacy-policy/8243040"
      onClick={[Function]}
    >
      Privacy Policy
    </a>
    Usage reporting:
    <label
      className="switch"
      style={
        Object {
          "marginLeft": "-5px",
          "marginTop": "-4px",
          "transform": "scale(0.75)",
        }
      }
    >
      <input
        checked={true}
        className="switch-input"
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="switch-label"
        data-off="Off"
        data-on="On"
      />
      <span
        className="switch-handle"
      />
    </label>
  </div>
</div>
`;

exports[`HistoryPanel renders correctly 1`] = `
<div
  className="history-list"
>
  <div
    className="panel-title"
  >
    <span
      aria-hidden={true}
      className="fa fa-history fa-fw"
    />
     History
  </div>
  <div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
    </div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abcd
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          efgh
        </span>
      </span>
    </div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          car
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          boat
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`HistoryPanel renders correctly 2`] = `
<div
  className="history-list"
>
  <div
    className="panel-title"
  >
    <span
      aria-hidden={true}
      className="fa fa-history fa-fw"
    />
     History
  </div>
  <div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abc
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          cdf
        </span>
      </span>
    </div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          abcd
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          efgh
        </span>
      </span>
    </div>
    <div
      className="history-list-item"
      onClick={[Function]}
    >
      <span>
        <span>
          car
        </span>
         
        <span
          aria-hidden={true}
          className="fa fa-long-arrow-right"
        />
         
        <span>
          boat
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`TemplatesPanel renders correctly 1`] = `
<div
  className="templates-list"
>
  <div
    className="panel-title panel-title-extended"
  >
    <span>
      <span
        aria-hidden={true}
        className="fa fa-file-text fa-fw"
      />
      <span
        style={
          Object {
            "paddingRight": "10px",
          }
        }
      >
         Templates
      </span>
      <span
        className="templates-panel-title-note-success"
      >
        (Click to paste)
      </span>
    </span>
    <span
      className="templates-create-new-button"
      onClick={[Function]}
    >
      <span
        aria-hidden={true}
        className="fa fa-plus"
        style={
          Object {
            "marginRight": "0.5em",
          }
        }
      />
      Create New
    </span>
  </div>
  <div>
    <div
      className="templates-list-item"
      onClick={[Function]}
    >
      <span
        title="Template: \\"cdf\\""
      >
        abc
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-pencil templates-list-item-edit"
          onClick={[Function]}
        />
        <span
          aria-hidden={true}
          className="fa fa-times templates-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="templates-list-item"
      onClick={[Function]}
    >
      <span
        title="Template: \\"This is a long text specified as template content, mouse hover tooltip should wrap the preview of this template.\\""
      >
        abc
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-pencil templates-list-item-edit"
          onClick={[Function]}
        />
        <span
          aria-hidden={true}
          className="fa fa-times templates-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="templates-list-item"
      onClick={[Function]}
    >
      <span
        title="Template: \\"Dalimil Hajek\\""
      >
        signature
      </span>
      <span>
        <span
          aria-hidden={true}
          className="fa fa-pencil templates-list-item-edit"
          onClick={[Function]}
        />
        <span
          aria-hidden={true}
          className="fa fa-times templates-list-item-remove"
          onClick={[Function]}
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`Whole widget renders correctly 1`] = `
<div
  className=""
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    className="main-panel"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <input
        className="text-input"
        name="findTextInput"
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Find"
        spellCheck="false"
        title={null}
        type="text"
        value=""
      />
      <div
        className="button-standard button-disabled button-small"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          aria-hidden={true}
          className="fa fa-chevron-up"
        />
      </div>
      <div
        className="button-standard button-disabled button-small"
        onClick={[Function]}
        style={Object {}}
      >
        <span
          aria-hidden={true}
          className="fa fa-chevron-down"
        />
      </div>
       
      <div
        className="search-status-text status-error"
      >
        No Target
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <input
        className="text-input"
        name="replaceTextInput"
        onChange={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        placeholder="Replace with"
        spellCheck="false"
        type="text"
        value=""
      />
      <div
        className="button-standard button-disabled"
        onClick={[Function]}
        style={Object {}}
      >
        Replace
      </div>
       
      <div
        className="button-standard button-disabled"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "0.5em",
          }
        }
      >
        Replace all
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "width": "50%",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
          title={null}
        >
          <label
            htmlFor="checkbox-matchCaseInput"
            style={
              Object {
                "cursor": "pointer",
                "order": "1",
              }
            }
          >
             
            Match Case
          </label>
          <div
            className="checkbox"
          >
            <input
              checked={false}
              id="checkbox-matchCaseInput"
              name="matchCaseInput"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="checkbox-matchCaseInput"
            />
          </div>
          <span
            className="checkbox-after-content"
            style={
              Object {
                "order": "2",
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
          title={null}
        >
          <label
            htmlFor="checkbox-wholeWordsInput"
            style={
              Object {
                "cursor": "pointer",
                "order": "1",
              }
            }
          >
             
            Whole Words
          </label>
          <div
            className="checkbox"
          >
            <input
              checked={false}
              id="checkbox-wholeWordsInput"
              name="wholeWordsInput"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="checkbox-wholeWordsInput"
            />
          </div>
          <span
            className="checkbox-after-content"
            style={
              Object {
                "order": "2",
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "width": "50%",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
          title={null}
        >
          <label
            htmlFor="checkbox-advancedSearchInput"
            style={
              Object {
                "cursor": "pointer",
                "order": "1",
              }
            }
          >
             
            Advanced Options
          </label>
          <div
            className="checkbox"
          >
            <input
              checked={false}
              id="checkbox-advancedSearchInput"
              name="advancedSearchInput"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              htmlFor="checkbox-advancedSearchInput"
            />
          </div>
          <span
            className="checkbox-after-content"
            style={
              Object {
                "order": "2",
              }
            }
          />
        </div>
        <div
          className="star-wrapper"
        >
          <span
            className="star-label"
            onClick={[Function]}
          >
             
            Save to Favourites
          </span>
          <span
            className="fa-stack fa-lg star-checkbox"
            onClick={[Function]}
          >
            <span
              aria-hidden={true}
              className="fa fa-star fa-stack-1x"
              style={
                Object {
                  "fontSize": "1.8em",
                }
              }
            />
            <span
              aria-hidden={true}
              className="fa fa-star-o fa-stack-2x"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="right-panel"
  >
    <div
      className="right-panel-content"
    />
    <div
      className="right-panel-buttons"
      onClick={[Function]}
    >
      <div
        className="menu-item"
        onClick={[Function]}
        title="Favourites"
      >
        <span
          aria-hidden={true}
          className="fa fa-star fa-2x fa-fw"
        />
      </div>
      <div
        className="menu-item"
        onClick={[Function]}
        title="History"
      >
        <span
          aria-hidden={true}
          className="fa fa-history fa-2x fa-fw"
        />
      </div>
      <div
        className="menu-item"
        onClick={[Function]}
        title="Templates"
      >
        <span
          aria-hidden={true}
          className="fa fa-book fa-2x fa-fw"
        />
      </div>
      <div
        className="menu-item"
        onClick={[Function]}
        title="Help"
      >
        <span
          aria-hidden={true}
          className="fa fa-question-circle fa-2x fa-fw"
        />
      </div>
    </div>
  </div>
</div>
`;
